stdlib/nodejs/file.rb

```diff
@@ -158,10 +158,15 @@ def self.symlink?(path)
     `return executeIOAction(function(){return __fs__.lstatSync(#{path}).isSymbolicLink()})`
   end
 
-  def self.expand_path(path, basedir = nil)
-    path = path.to_str
+  def self.absolute_path(path, basedir = nil)
+    path = path.respond_to?(:to_path) ? path.to_path : path
     basedir ||= Dir.pwd
-    `return __path__.normalize(__path__.resolve(#{basedir.to_str}, #{path})).split(__path__.sep).join(__path__.posix.sep)`
+    `return __path__.normalize(__path__.resolve(#{basedir.to_str}, #{path.to_str})).split(__path__.sep).join(__path__.posix.sep)`
+  end
+
+  def self.expand_path(path, basedir = nil)
+    # we should resolve ~ before calling absolute_path! see https://github.com/opal/opal/pull/1924
```

Comment:
Have you tried removing `expand_path` from the node version and relying on the corelib implementation? the platform-specific code seems to be all inside `absolute_path`

Reply:
No but that's a good idea 👍